src/callbacks/callbacks.jl

```diff
@@ -168,15 +168,34 @@ NVTX.@annotate function rrtmgp_model_callback!(integrator)
     if !(radiation_mode isa RRTMGPI.GrayRadiation)
         if radiation_mode.aerosol_radiation
             ᶜΔz = Fields.Δz_field(Y.c)
-            ᶜaero_type =
-                Fields.array2field(rrtmgp_model.center_aerosol_type, axes(Y.c))
-            @. ᶜaero_type =
-                set_aerosol_type(p.tracers.prescribed_aerosols_field)
-            ᶜaero_conc = Fields.array2field(
-                rrtmgp_model.center_aerosol_column_mass_density,
-                axes(Y.c),
-            )
-            @. ᶜaero_conc = maximum(p.tracers.prescribed_aerosols_field) * ᶜΔz
+
+            aerosol_names_pair = [
+                (:center_dust_column_mass_density, :DST01),
+                (:center_ss_column_mass_density, :SSLT01),
+                (:center_so4_column_mass_density, :SO4),
+                (:center_bcpi_column_mass_density, :CB2),
+                (:center_bcpo_column_mass_density, :CB1),
+                (:center_ocpi_column_mass_density, :OC2),
+                (:center_ocpo_column_mass_density, :OC1),
+            ]
+
+            for (aerosol_name, prescribed_aerosol_name) in aerosol_names_pair
+                ᶜaero_conc = Fields.array2field(
+                    getproperty(rrtmgp_model, aerosol_name),
+                    axes(Y.c),
+                )
+                if prescribed_aerosol_name in
+                   propertynames(p.tracers.prescribed_aerosols_field)
+                    @. ᶜaero_conc =
```

Comment:
This doesn't work on GPU: https://buildkite.com/clima/climaatmos-ci/builds/20263#01919159-513a-4f3e-bb65-44ec284ec892/205-300. @Sbozzolo @dennisYatunin Is there a way to make it work on GPU (while I can still understand the code...)?